### ISSUE TITLE
Octave - no syntax highligting

### ISSUE THREAD

**Issue by hajgato** [hpcugent/easybuild-easyconfigs#2767]
`Octave` needs `QScintilla2` to have syntax highlighting. `QScintilla2` depends on `Qt` (can be used with both `Qt4` and `Qt5`, and it installs itself into the original `Qt` directory. I did not investigate how could I install the necessary files to another directory, and if it is installed in another directory, how could be used. Its the fascinating world of `qmake`

**Comment by ocaisa** [comment]
I got around this by adding Qscintilla to the initial Qt install. The relevant bits are:
```
#QScintilla
qscintilla = 'QScintilla'
qscintilla_version = '2.9.2'

source_urls = [
    'http://download.qt.io/official_releases/qt/%(version_major_minor)s/%(version)s/',
    'http://download.qt.io/archive/qt/%(version_major_minor)s/%(version)s/',
    #Qscintilla
    'http://sourceforge.net/projects/pyqt/files/QScintilla2/%s-%s/' % (qscintilla, qscintilla_version),
]
sources = [
    '%(namelower)s-everywhere-opensource-src-%(version)s.tar.gz',
    '%s_gpl-%s.tar.gz' % (qscintilla, qscintilla_version),
]
```
and
```
# Install QScintilla (needed by Octave) by brute force
installqscintilla = "cd %(builddir)s &&"
installqscintilla += "cd %s_gpl-%s/Qt4Qt5 &&" % (qscintilla, qscintilla_version)
installqscintilla += "%(installdir)s/bin/qmake qscintilla.pro && make && make install"
postinstallcmds = [installqscintilla]
```

**Comment by hajgato** [comment]
@ocaisa Where does it install `QScintilla`? To the origininal `Qt` directory, or somewhere else? I am not so happy if it installs to the original `Qt` directory.

**Comment by ocaisa** [comment]
Yeah, its a hack, it installs at the same time as Qt so adds the libs to the same directory

**Comment by boegel** [comment]
@ocaisa any reason why you're not including `QScintilla` as a dependency?

**Comment by ocaisa** [comment]
Because I don't know how to install Qt extensions without them ending up in the initial Qt install path.

On 2 May 2016 5:12 p.m., "Kenneth Hoste" <notifications@github.com<mailto:notifications@github.com>> wrote:

@ocaisa<https://github.com/ocaisa> any reason why you're not including QScintilla as a dependency?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/hpcugent/easybuild-easyconfigs/issues/2767#issuecomment-216246139>


------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------
Forschungszentrum Juelich GmbH
52425 Juelich
Sitz der Gesellschaft: Juelich
Eingetragen im Handelsregister des Amtsgerichts Dueren Nr. HR B 3498
Vorsitzender des Aufsichtsrats: MinDir Dr. Karl Eugen Huthmacher
Geschaeftsfuehrung: Prof. Dr.-Ing. Wolfgang Marquardt (Vorsitzender),
Karsten Beneke (stellv. Vorsitzender), Prof. Dr.-Ing. Harald Bolt,
Prof. Dr. Sebastian M. Schmidt
------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------

**Comment by hajgato** [comment]
@boegel @ocaisa 
I tried to look into this issue at the beginning, and I found that it is not possible to install `Qt` extensions to a separate directory without patching `Qt`. At least I was not able to figure out, although I did not spend too much time with this issue, as it was not crucial for me.
